src/debugger/debugging.ts

```diff
@@ -0,0 +1,88 @@
+import * as vscode from 'vscode';
+import {
+    doCheckModel, getSpecFiles
+} from '../commands/checkModel';
+
+export const TLAPLUS_DEBUG_LAUNCH_CHECKNDEBUG = 'tlaplus.debug.checkAndDebugEditorContents';
+export const TLAPLUS_DEBUG_LAUNCH_DEBUG = 'tlaplus.debug.debugEditorContents';
+
+export class TLADebugAdapterServerDescriptorFactory implements vscode.DebugAdapterDescriptorFactory {
+
+    createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined):
+        vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+        return new vscode.DebugAdapterServer(session.configuration['port']);
+    }
+}
+
+/**
+ * Attaches the DAP front-end to an already running TLC debugger.
+ */
+export async function debugSpec(
+    resource: vscode.Uri | undefined,
+    diagnostic: vscode.DiagnosticCollection,
+    context: vscode.ExtensionContext
+): Promise<void> {
+    let targetResource = resource;
+    if (!targetResource && vscode.window.activeTextEditor) {
+        // Since this command is registered as a button on the editor menu, I don't
+        // think this branch is ever taken.  It's here because the DAP example has it.
+        targetResource = vscode.window.activeTextEditor.document.uri;
+    }
+    if (targetResource) {
+        // Attaching to a separately launched TLC leaves the result view (webview) empty.
+        // However, TLC sends its output via the DAP Output event
+        // (https://microsoft.github.io/debug-adapter-protocol/specification#Events_Output)
+        // to VSCode.  Somebody has to figure out how to wire this up. In the meantime,
+        // users have to manually "parse" the output on VSCode's DebugConsole. Unfortunately,
+        // it is TLC's '-tool' output containing "@!@!@" markers around each message.
```

Comment:
It's possible to capture such debug messages by means of a `DebugAdapterTracker` and `debug.registerDebugAdapterTrackerFactory()`. But in this scenario we connect to already running TLC which means we'll skip some messages at the beginning. Without having all the TLC messages, it's impossible to display the TLC check result properly.

Though, there's an option to implement the feature even for an already running process: if TLC could keep all its output and return it on a custom request from the debug adapter, the extension would send such request on attaching and obtain all the necessary information for displaying the result view.

Reply:
IIRC I implemented buffering in the TLA+ debugger.  That said, I don't think that the web view is super important while debugging.

Reply:
> I don't think that the web view is super important while debugging

I agree. But again, the comment suggests to find a way to do that.

Reply:
I'd like to postpone implementing this functionality until after this PR. :-)

Reply:
Sure, I also don't think it should be implemented now (or even at all).